src/reconcile/rec_visibility.c

```diff
@@ -340,6 +339,17 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
             }
         }
 
+        /*
+         * Handle double tombstone scenario that can happen in the history store when we add zero
+         * timestamped tombstones to the front of the update chain.
+         */
+        if (upd->type == WT_UPDATE_TOMBSTONE && upd->next != NULL &&
```

Comment:
Can we also assert it only happens in history store?

Reply:
We should do this only when the first update is not globally visible.

Reply:
Otherwise, we will pin it forever in cache.

Reply:
That's why I like it to be outside the loop to avoid calling global visible check in the loop.

Reply:
Yeah I'll move it further down

Reply:
It'll still get pinned until its globally visible though, right?

Reply:
Yes

Reply:
Interestingly enough we hit this assert very frequently, suggesting double tombstones aren't a history store only issue.

Reply:
Check if upd->next is aborted?